platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -49,7 +70,118 @@ public static String formatNodeName(final long nodeId) {
      * @return its Hash
      */
     @NonNull
-    public static Hash hash(@NonNull final Roster roster) {
+    public static Hash hashOf(@NonNull final Roster roster) {
         return PBJ_RECORD_HASHER.hash(roster, Roster.PROTOBUF);
     }
+
+    /**
+     * Generates the initial active roster based on the given software version and initial state.
+     * The active roster is obtained by adopting the candidate roster if a software upgrade is detected.
+     * Otherwise, the active roster is created from the address book.
+     *
+     * @param version the software version of the current node
+     * @param initialState the initial state of the platform
+     * @param addressBook  the address book being used by the network
+     * @return the active roster which will be used by the platform
+     */
+    public static @NonNull Roster generateActiveRoster(
+            @NonNull final SoftwareVersion version,
+            @NonNull final ReservedSignedState initialState,
+            @NonNull final AddressBook addressBook) {
+        final boolean softwareUpgrade = detectSoftwareUpgrade(version, initialState.get());
+        final MerkleRoot merkleRoot = initialState.get().getState();
+        final WritableRosterStore rosterStore = merkleRoot.getWritableRosterStore();
+        final Roster candidateRoster = rosterStore.getCandidateRoster();
+
+        return determineRoster(
```

Comment:
I still don't see the point in extracting this as a separate method. Why not put the contents of `determineRoster()` here in this method?

Reply:
Done.